crates/engine/tree/src/tree/root.rs

```diff
@@ -0,0 +1,60 @@
+//! State root task related functionality.
+
+use reth_provider::providers::ConsistentDbView;
+use reth_trie::{updates::TrieUpdates, TrieInput};
+use reth_trie_parallel::parallel_root::ParallelStateRootError;
+use revm_primitives::{EvmState, B256};
+use std::{
+    future::Future,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+use tokio_stream::wrappers::UnboundedReceiverStream;
+
+/// Standalone task that receives a transaction state stream and updates relevant
+/// data structures to calculate state root.
+///
+/// It is responsile of  initializing a blinded sparse trie and subscribe to
+/// transaction state stream. As it receives transaction execution results, it
+/// fetches the proofs for relevant accounts from the database and reveal them
+/// to the tree.
+/// Then it updates relevant leaves according to the result of the transaction.
+#[allow(dead_code)]
+pub(crate) struct StateRootTask<Factory> {
+    /// View over the state in the database.
+    consistent_view: ConsistentDbView<Factory>,
```

Comment:
Is this still something we want to retain in our new designs? I thought the consistent db viewer was a 'hack' and we were looking for ways to do this better? Maybe I misremember